fabric-item-group-api-v1/src/main/java/net/fabricmc/fabric/mixin/itemgroup/ItemGroupMixin.java

```diff
@@ -18,42 +18,48 @@
 
 import java.util.LinkedList;
 import java.util.Objects;
+import java.util.UUID;
 
-import org.objectweb.asm.Opcodes;
-import org.spongepowered.asm.mixin.Final;
+import org.jetbrains.annotations.Nullable;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
-import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemGroups;
 import net.minecraft.item.ItemStackSet;
 import net.minecraft.resource.featuretoggle.FeatureSet;
 import net.minecraft.util.Identifier;
 
 import net.fabricmc.fabric.api.event.Event;
-import net.fabricmc.fabric.api.itemgroup.v1.IdentifiableItemGroup;
 import net.fabricmc.fabric.api.itemgroup.v1.FabricItemGroupEntries;
+import net.fabricmc.fabric.api.itemgroup.v1.IdentifiableItemGroup;
 import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents;
+import net.fabricmc.fabric.impl.itemgroup.FabricItemGroup;
 import net.fabricmc.fabric.impl.itemgroup.ItemGroupEventsImpl;
 import net.fabricmc.fabric.impl.itemgroup.MinecraftItemGroups;
 
 @Mixin(ItemGroup.class)
-abstract class ItemGroupMixin implements IdentifiableItemGroup {
-	@Shadow
-	@Final
-	private int index;
-
+abstract class ItemGroupMixin implements IdentifiableItemGroup, FabricItemGroup {
 	@Shadow(aliases = "field_40859")
 	private ItemStackSet displayStacks;
 
 	@Shadow(aliases = "field_40860")
 	private ItemStackSet searchTabStacks;
 
+	@Unique
+	private int page = -1;
+
+	@Unique
+	@Nullable
+	private UUID fallbackUUIDCache;
```

Comment:
Make it more apparent what mod the mixin came from
```suggestion
	@Unique
	private int fabric_page = -1;

	@Unique
	@Nullable
	private UUID fabric_fallbackUUIDCache;
```

Reply:
and all references need changing as well